src/services/WorkspaceFlowRunsApi.ts

```diff
@@ -72,16 +70,40 @@ export class WorkspaceFlowRunsApi extends WorkspaceApi implements IWorkspaceFlow
     return data
   }
 
-  public async getFlowRunsGraph(flowRunId: string): Promise<GraphNode[]> {
-    const { data } = await this.get<FlowRunGraphResponse[]>(`/${flowRunId}/graph`)
-
-    return mapper.map('FlowRunGraphResponse', data, 'GraphNode')
-  }
-
-  public async getFlowRunsTimeline(id: string): Promise<GraphTimelineNode[]> {
-    const { data } = await this.get<FlowRunGraphResponse[]>(`/${id}/graph`)
+  public async getFlowRunsGraph(id: string): Promise<RunGraphData> {
+    // todo: should we just send this through the mapper?
+    // context: the idea is that this would be more efficient since the response can be very large.
+    // But that is unproven from a perf perspective. However this is much simpler than a dedicated mapper.
+    // But generally consistency is better so should test if this even makes a difference perf wise.
+    const reviver = (key: string, value: unknown): unknown => {
+      if (key === 'start_time' || key == 'end_time') {
+        if (typeof value !== 'string') {
+          return value
+        }
+
+        const date = parseISO(value)
+
+        if (isValid(date)) {
+          return date
+        }
+      }
+
+      if (key === 'nodes') {
+        if (!Array.isArray(value)) {
+          return
+        }
+
+        return new Map(value)
+      }
+
+      return value
+    }
+
+    const { data } = await this.get<RunGraphData>(`/${id}/graph-v2`, {
+      transformResponse: response => JSON.parse(response, reviver),
+    })
```

Comment:
For consistency I think this should end up in the mapper - it's a little different from the way we map other payloads but I think the idea is the same - take an API-readable response and transform it into a UI-readable response.

Reply:
Yeah this approach was strictly for performance because this payload can have up to 10k nodes. But I agree the performance gains are likely minimal and not worth the inconsistency. 

I'll update this in a follow up PR.